app/controllers/api/v0/sources_controller.rb

```diff
@@ -6,21 +6,27 @@ class SourcesController < ApplicationController
       include Api::Mixins::ShowMixin
 
       def create
-        body = JSON.parse(request.body.read)
-        source = Source.create!(:tenant_id => body["tenant_id"], :name => body["name"])
+        source = Source.create!(create_params)
         render :json => source, :status => :created, :location => api_v0x0_source_url(source.id)
       end
 
-
       def update
-        Source.update(params[:id], :name => params[:name])
+        Source.update(params.require(:id), update_params)
         head :no_content
       rescue ActiveRecord::RecordNotFound
         head :not_found
       end
 
       private
 
+      def create_params
+        ActionController::Parameters.new(JSON.parse(request.body.read)).permit(:name, :tenant_id)
```

Comment:
Not for this PR, but this pattern feels like something we could provide in ApplicationController...

```ruby
def body_params
  @body_params ||= ActionController::Parameters.new(JSON.parse(request.body.read))
end
```